tools/capture_radar.rb

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env ruby
+require File.expand_path("../config/environment", __dir__)
+require 'trollop'
+
+opts = Trollop.options(ARGV) do
+  banner "USAGE:   #{__FILE__} -h <number of hours back to query metrics>\n" \
+         "Example: #{__FILE__} -d <number of days back to query metrics>"
+
+  opt :hours, "Hours", :short => "h", :type => :int, :default => 4
+  opt :days,  "Days",  :short => "d", :type => :int
+  opt :label, "Label", :short => "l", :type => :string, :default => "com.redhat.component"
+end
+Trollop.die :hours, "is required" unless opts[:hours] || opts[:days_given]
+
+Metric
+class Metric
+  def derived_cpu_total_cores_used
+    return if cpu_usage_rate_average.nil? || derived_vm_numvcpus.nil? || derived_vm_numvcpus == 0
+    (cpu_usage_rate_average * derived_vm_numvcpus) / 100.0
+  end
+end
+
+CustomAttribute
+class CustomAttribute
+  has_many   :metric_rollups, :as => :resource
+end
+
+TIME_RANGE = if opts[:days_given]
+               [opts[:days].days.ago.utc.beginning_of_hour..Time.now.utc.end_of_hour]
+             else
+               [opts[:hours].hours.ago.utc.beginning_of_hour..Time.now.utc.end_of_hour]
+             end
+
+labels = CustomAttribute.where(:section => ['labels', 'docker_labels'], :resource_type => ['ContainerImage', 'ContainerGroup'], :name => opts[:label])
+
+# Question - If a container is short lived, will we still be able to find it here?
+labels_containers = labels.each_with_object(Hash.new { |h,k| h[k] = [] }) { |l, h| h[l] += l.resource.containers.to_a }
+
+# Group metrics by label/project
+labels_metrics = Hash.new { |h,k| h[k] = [] }
+labels_containers.each do |l, containers|
+  containers.each do |c|
+    labels_metrics[[l, c.container_project]] += c.metrics.where(:timestamp => TIME_RANGE).to_a # This should be scoped to time range for the hour
+  end
+end
```

Comment:
@gtanzillo @Fryguy so, scenario

20k active containers, with 4 hours of metrics, 20k archived containers with no metrics. For simplicity each of the container has unique image and each of them matches label we observe. We can do other scenarios with more containers per 1 image.

Results
========

after 1h, It hasn't passed through this line and it takes funky 40GB in memory (and rising) :-)